src/BuildScriptGenerator/DefaultCompatiblePlatformDetector.cs

```diff
@@ -134,22 +133,42 @@ public class DefaultCompatiblePlatformDetector : ICompatiblePlatformDetector
 
                 _logger.LogDebug($"No platform version found, " +
                                  $"setting to the detected version '{detectionResult.LanguageVersion}'.");
-                platformVersion = detectionResult.LanguageVersion;
+                detectedPlatformVersion = detectionResult.LanguageVersion;
+
+                platformResult = Tuple.Create(selectedPlatform, detectedPlatformVersion);
+                _logger.LogDebug($"Detected platform '{platformName}' with version '{detectedPlatformVersion}'.");
+            }
+            else
+            {
+                platformResult = Tuple.Create(selectedPlatform, platformVersionFromOptions);
             }
 
-            _logger.LogDebug($"Detected platform '{platformName}' with version '{platformVersion}'.");
-            platformResult = Tuple.Create(selectedPlatform, platformVersion);
             return true;
         }
 
         private bool IsEnabledForMultiPlatformBuild(IProgrammingPlatform platform, RepositoryContext ctx)
         {
-            if (ctx.DisableMultiPlatformBuild)
+            if (_commonOptions.EnableMultiPlatformBuild)
             {
-                return false;
+                return platform.IsEnabledForMultiPlatformBuild(ctx);
             }
 
-            return platform.IsEnabledForMultiPlatformBuild(ctx);
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the platform version in a hierarchical fasion
+        /// 1. --platform nodejs --platform-version 4.0
+        /// 2. NODE_VERSION=4.0 from environment variables
+        /// 3. NODE_VERSION=4.0 from build.env file
+        /// </summary>
+        /// <param name="platformName">Platform for which we want to get the version in a hierarchical way.</param>
+        /// <returns></returns>
+        private string GetPlatformVersion(string platformName)
+        {
+            platformName = platformName == "nodejs" ? "node" : platformName;
```

Comment:
Ugh, this is one more item for 'clean-up' task.  This has caused us unnecessary cycles. Let us discuss in scrum meeting.